### ISSUE TITLE
Avro format is not supported

### ISSUE THREAD

**Issue by NikNikM** [fluent/fluent-bit#4097]
## Bug Report

**Describe the bug**
The Kafka output plugin is not supporting **avro** in the _format_ attribute:
```
[OUTPUT]
    Name kafka
    Match       *
    format avro
    ...
```

**To Reproduce**
- Configure the Kafka output to send the messages using the config above
- Messages are sent using the default json format

**Expected behavior**
In the release notes of the version 1.7: https://fluentbit.io/announcements/v1.7.0/
I've seen the avro support: **avro: new avro interface and unit tests**

**Your Environment**

* Version used: 1.8.6
* Configuration: above

**Additional context**
1) We tried to use http output plugin with custom headers:
    header Content-Type application/vnd.kafka.avro.v2+json
    header Accept application/vnd.kafka.v2+json, application/vnd.kafka+json, application/json
  The Kafka Rest Proxy accepts the list of records in the following format:
  {
  "records": [
    {"key": "value"},
    {"key": "value"},
    ...
   ]}
   and this is simply not possible to configure as chunks are made using msgpack, json,  json_stream, json_lines and gelf.
   
2) Kafka Rest output plugin doesn't support Avro format as there is no way to specify the custom header used in http output plugin.

**Comment by nokute78** [comment]
https://github.com/fluent/fluent-bit/blob/v1.8.6/CMakeLists.txt#L100
Hmm, a build configuration `FLB_AVRO_ENCODER` is disabled.

**Comment by NikNikM** [comment]
Thank you for your message. Is this going to be available in the next version (1.8.8)?

**Comment by NikNikM** [comment]
Ok, I tried to use it and there are some problems :(
1) Type long is automatically parsed as int: https://github.com/fluent/fluent-bit/blob/v1.8.6/src/flb_avro.c#L89
Avro schema: `"fields":[{"name":"timestamp","type":{"type":"long"} ...`
2) Type double is not parsed; in my case, I have `"value": 5` that I would like to write as a double. It is parsed as int and it fails as the schema is different: `{"name":"value","type":"double"...`. Even if I convert it with Lua script to 5.0, it's still int

**Comment by nokute78** [comment]
Thank you for pointing.

1. Should we use `avro_value_set_long` anytime? Currently we use msgpack-c which treats `MSGPACK_OBJECT_POSITIVE_INTEGER` as 64bit integer. It doesn't support 8/16/32 bit integer types.
2. What is input plugin ? or If you use parser , you can use `types` property to convert data type. https://docs.fluentbit.io/manual/v/1.3/parser#configuration-parameters

**Comment by NikNikM** [comment]
1. I tested it and it works in my case (as in my Avro schema I have only longs (no integers)) but I think it won't be solution that works for all cases; integers and longs should be threated separately.
2. This is a great tip! We are using tail and regex parsers. In the meantime, we switched from kafka_out to kafka_rest_out and I did a change in order to be able to support Avro format (adding new header to tell Kafka Rest API that the message should be stored in Avro format): https://github.com/fluent/fluent-bit/pull/4113

Thanks again for your help
